src/checkbox/checkbox.ts

```diff
@@ -0,0 +1,100 @@
+import {Component, ElementRef, EventEmitter, forwardRef, Input, Output} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+
+const NGB_CHECKBOX_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => NgbCheckbox),
+  multi: true
+};
+
+let nextId = 1;
+
+/**
+ * A component that allows the user to edit boolean values.
+ */
+@Component({
+  selector: 'ngb-checkbox',
+  host: {'class': 'custom-control custom-checkbox'},
+  template: `
+    <input
+      type="checkbox"
+      class="custom-control-input"
+      [disabled]="disabled || readonly"
```

Comment:
Not sure here, the `readonly` is really a different beast from `disabled`. 
in NG Material it's used to control the 'can one change value if control is readonly':
https://github.com/angular/components/blob/8f62ffe7359d929ab28bfc6fa8bf4fafcebc3dc6/src/material/input/input.ts#L82
What's your stance on this?

Reply:
I agree with you, in our case the readonly flag is only used for "disabled" state of the underlying input which in turn will properly style it.
I guess it would be more semantically correct to treat readonly as its own. I'll try something